### PR TITLE
csi: propagate error messages

### PR DESCRIPTION
**Pull request by raunakkumar**
Signed-off-by: Raunak <raunak.kumar@hpe.com>

### CODE REVIEW

File: pkg/driver/controller_server.go [hunk 1]
```diff
@@ -224,7 +224,7 @@ func (driver *Driver) CreateVolume(ctx context.Context, request *csi.CreateVolum
 	createParameters, err := driver.flavor.ConfigureAnnotations(request.Name, request.Parameters)
 	if err != nil {
 		log.Errorf("Failed to configure create parameters from PVC annotations. err=%v", err)
-		return nil, status.Error(codes.Internal, "Failed to configure create parameters from PVC annotations")
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to configure create parameters from PVC annotations, %s", err.Error()))
```
Comment (rkumpf) [inline pkg/driver/controller_server.go:227]: I haven't seen a comma used as a message separator like this.  A colon is often used, and sometimes a period.  I suggest choosing between those two.

**Review by rkumpf — APPROVED** [review]

File: pkg/driver/controller_server.go (hunk 1 above)
Comment (raunakkumar) [reply pkg/driver/controller_server.go]: We have it at many places in the file, but sure will update.

**Review by rkumpf — APPROVED** [review]

File: pkg/driver/controller_server.go [hunk 2]
```diff
@@ -408,7 +408,7 @@ func (driver *Driver) createVolume(
 	existingVolume, err := storageProvider.GetVolumeByName(name)
 	if err != nil {
 		log.Error("err: ", err.Error())
-		return nil, status.Error(codes.Unavailable, "Failed to check if volume exists")
+		return nil, status.Error(codes.Unavailable, fmt.Sprintf("Failed to check if volume exists. %s", err.Error()))
```
Comment (suneeth51) [inline pkg/driver/controller_server.go:411]: nit: Do you want to log volume name as well? If its already captured in the error message, feel free to skip this comment.

**Review by suneeth51 — APPROVED** [review]
lgtm..

**Pull request merged**